src/ages.ts

```diff
@@ -0,0 +1,105 @@
+        initialTimestamp: BigNumber.from(new Date("2022-06-08T17:00:06.000Z").getTime() / 1000),
+        initialBlock: 14_927_832,
+        finalTimestamp: BigNumber.from(new Date("2022-07-13T17:00:00.000Z").getTime() / 1000),
+        finalBlock: 15_135_480,
+        totalEmission: BigNumber.from(350_000),
+        epochName: "epoch1",
+      },
+      epoch2: {
+        snapshotBlock: 15_134_933, // https://etherscan.io/block/15134933
+        initialTimestamp: BigNumber.from(new Date("2022-07-13T17:00:06.000Z").getTime() / 1000),
+        initialBlock: 15_135_481,
+        finalTimestamp: BigNumber.from(new Date("2022-08-16T17:00:00.000Z").getTime() / 1000),
+        finalBlock: 15_353_545,
+        totalEmission: BigNumber.from(1_700_000),
+        epochName: "epoch2",
+      },
+      epoch3: {
+        snapshotBlock: 15_353_032, // https://etherscan.io/block/15353032
+        initialTimestamp: BigNumber.from(new Date("2022-08-16T17:00:06.000Z").getTime() / 1000),
+        initialBlock: 15_353_547,
+        finalTimestamp: BigNumber.from(new Date("2022-09-20T15:00:00.000Z").getTime() / 1000), // 17h CET
+        finalBlock: undefined,
+        totalEmission: BigNumber.from(2_950_000),
+        epochName: "epoch3",
+      },
+    },
+  },
+  ["age2"]: {
+    ageName: "age2",
+    startTimestamp: BigNumber.from(new Date("2022-09-20T15:00:00.000Z").getTime() / 1000),
+    endTimestamp: BigNumber.from(new Date("2022-12-20T15:00:00.000Z").getTime() / 1000),
+    distribution: ageOneDistribution,
+    subgraphUrl: "https://api.thegraph.com/subgraphs/name/morpho-dev/morpho-rewards-staging",
+
+    epochs: {
+      epoch1: {
+        snapshotBlock: 14_927_832, // https://etherscan.io/block/14927832
+        initialTimestamp: BigNumber.from(new Date("2022-09-20T15:00:00.000Z").getTime() / 1000),
+        initialBlock: 14_927_832,
+        finalTimestamp: BigNumber.from(new Date("2022-10-20T15:00:00.000Z").getTime() / 1000),
+        finalBlock: 15_135_480,
+        totalEmission: BigNumber.from(350_000),
+        epochName: "epoch1",
+      },
+      epoch2: {
+        snapshotBlock: 15_134_933, // https://etherscan.io/block/15134933
+        initialTimestamp: BigNumber.from(new Date("2022-10-20T15:00:00.000Z").getTime() / 1000),
+        initialBlock: 15_135_481,
+        finalTimestamp: BigNumber.from(new Date("2022-11-20T15:00:00.000Z").getTime() / 1000),
+        finalBlock: 15_353_545,
+        totalEmission: BigNumber.from(1_700_000),
+        epochName: "epoch2",
+      },
+      epoch3: {
+        snapshotBlock: 15_353_032, // https://etherscan.io/block/15353032
+        initialTimestamp: BigNumber.from(new Date("2022-11-20T15:00:00.000Z").getTime() / 1000),
+        initialBlock: 15_353_547,
+        finalTimestamp: BigNumber.from(new Date("2022-12-20T15:00:00.000Z").getTime() / 1000), // 17h CET
+        finalBlock: undefined,
+        totalEmission: BigNumber.from(2_950_000),
+        epochName: "epoch3",
+      },
```

Comment:
Not really, because the final timestamp is computed before a block was mined at this timestamp, and there is not necessarily a block with this exact timestamp. And we use the timestamp at the beginning of the epoch to compute the rate of the given market.

Reply:
Let's discuss about it face 2 face!